nautobot/extras/forms/forms.py

```diff
@@ -982,9 +986,31 @@ class Meta:
             "has_sensitive_variables",
             "job_queues_override",
             "job_queues",
+            "default_job_queue_override",
+            "default_job_queue",
             "tags",
         ]
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        job_model = self.instance
+        if job_model.default_job_queue:
+            default_job_queue = job_model.default_job_queue.pk
+        else:
+            # Populate the field with settings.CELERY_TASK_DEFAULT_QUEUE JobQueue if it exists
+            try:
+                default_job_queue = JobQueue.objects.get(name=settings.CELERY_TASK_DEFAULT_QUEUE)
+                default_job_queue = default_job_queue.pk
+            except JobQueue.DoesNotExist:
+                pass
+        self.fields["default_job_queue"] = DynamicModelChoiceField(
+            queryset=JobQueue.objects.all(),
+            required=False,
+            help_text="The default job queue to route this job to",
+            label="Default Job queue",
+        )
+        self.fields["default_job_queue"].initial = default_job_queue
```

Comment:
This feels over-complex to me. Can we maybe instead design things so that `Job.default_job_queue` is non-nullable (with a on_delete=PROTECT) so that we always have one available, instead of needing this kind of fallback logic?